cuBERT_topic_modelling/cuBERTopic.py

```diff
@@ -0,0 +1,485 @@
+
+        Returns:
+            info: The information relating to all topics
+        """
+
+        # Note: getting topics in sorted order without using
+        # TopicMapper, as in BERTopic
+        test = self.topic_sizes_df.Name.str.split(
+            "_", expand=True
+        )[[0, 1, 2, 3, 4]]
+        self.original_topic_mapping = test[0]
+        self.new_topic_mapping = sorted(
+            self.topic_sizes_df["Topic"].to_pandas()
+        )
+
+        self.original_topic_mapping = self.original_topic_mapping.to_arrow().to_pylist()
+        self.final_topic_mapping = dict(zip(self.new_topic_mapping, 
+                                            self.original_topic_mapping))
+        test[0] = self.new_topic_mapping
+        test["Name"] = (
+            test[0].astype("str")
+            + "_"
+            + test[1]
+            + "_"
+            + test[2]
+            + "_"
+            + test[3]
+            + "_"
+            + test[4]
+        )
+        self.topic_sizes_df["Name"] = test["Name"]
+        self.topic_sizes_df["Topic"] = test[0]
+        return self.topic_sizes_df
+
+    def update_topic_size(self, documents):
+        """Calculate the topic sizes
+        Arguments:
+            documents: Updated dataframe with documents and
+            their corresponding IDs and newly added Topics
+        """
+
+        topic_sizes = (
+            documents.groupby(["Topic"])
+            .Document.count()
+            .reset_index()
+            .rename({"Topic": "Topic", "Document": "Count"}, axis="columns")
+            .sort_values("Count", ascending=False)
+        )
+        self.topic_sizes_df = topic_sizes
+
+    def sort_mappings_by_frequency(self, documents):
+        """Reorder mappings by their frequency.
+        For example, if topic 88 was mapped to topic
+        5 and topic 5 turns out to be the largest topic,
+        then topic 5 will be topic 0. The second largest,
+        will be topic 1, etc.
+        If there are no mappings since no reduction of topics
+        took place, then the topics will simply be ordered
+        by their frequency and will get the topic ids based
+        on that order.
+        This means that -1 will remain the outlier class, and
+        that the rest of the topics will be in descending order
+        of ids and frequency.
+        Arguments:
+            documents: Dataframe with documents and their
+            corresponding IDs and Topics
+        Returns:
+            documents: Updated dataframe with documents and the mapped
+                       and re-ordered topic ids
+        """
+
+        self.update_topic_size(documents)
+        # Map topics based on frequency
+        df = (
+            cudf.DataFrame(
+                self.topic_sizes_df.to_pandas().to_dict(),
+                columns=["Old_Topic", "Size"]
+            )
+            .sort_values("Size", ascending=False)
+            .to_pandas()
```

Comment:
Why are we converting to pandas here.  Is it cause of needing map. I think we can write an equivalent using map primitives.

Reply:
It is because of the `zip` operation, since we can't iterate on a `cudf.Series`, I think it would be great if we can rewrite this part of the code!

Reply:
Still pending.

Reply:
Still pending.